File: Python/reverse_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev_ptr, curr_ptr = None, head

        while curr_ptr:
            temp_list = curr_ptr.next
            curr_ptr.next = prev_ptr
            prev_ptr = curr_ptr
            curr_ptr = temp_list
        return prev_ptr

File: Cpp/reverse_linked_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev_ptr = NULL;
        ListNode* curr_ptr = head;
        while(curr_ptr != NULL){
            ListNode* next = curr_ptr -> next;
            curr_ptr -> next = prev_ptr;
            prev_ptr = curr_ptr;
            curr_ptr = next;
        }
        return prev_ptr;        
    }
};